app/routes/synonyms._index.tsx

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { LoaderFunctionArgs } from '@remix-run/node'
+import {
+  Form,
+  Link,
+  useLoaderData,
+  useSearchParams,
+  useSubmit,
+} from '@remix-run/react'
+import {
+  Button,
+  Grid,
+  GridContainer,
+  Icon,
+  Label,
+  TextInput,
+} from '@trussworks/react-uswds'
+import { useId, useState } from 'react'
+
+import type { SynonymGroup } from './synonyms/synonyms.lib'
+import { searchSynonymsByEventId } from './synonyms/synonyms.server'
+import { ToolbarButtonGroup } from '~/components/ToolbarButtonGroup'
+import PaginationSelectionFooter from '~/components/pagination/PaginationSelectionFooter'
+
+import searchImg from 'nasawds/src/img/usa-icons-bg/search--white.svg'
+
+export async function loader({ request: { url } }: LoaderFunctionArgs) {
+  const { searchParams } = new URL(url)
```

Comment:
Probably should check that the user is a moderator in the loader here.

Reply:
That is done at the parent route level [here](https://github.com/nasa-gcn/gcn.nasa.gov/blob/main/app/routes/synonyms/route.tsx#L18). I'm happy to also add it here, but I don't think it's necessary because the entire synonyms route is mod gated.

Reply:
I think it is necessary. As far as I can recall, Remix will request the loader data for all of the ancestor routes in parallel, so authorization-dependent code _must_ be checked even in descendent routes.